catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/formulaeditor/ColorPickerParameterizedTest.java

```diff
@@ -0,0 +1,133 @@
+import org.catrobat.catroid.uiespresso.content.brick.utils.BrickTestUtils;
+import org.catrobat.catroid.uiespresso.content.brick.utils.CustomSwipeAction;
+import org.catrobat.catroid.uiespresso.testsuites.Cat;
+import org.catrobat.catroid.uiespresso.testsuites.Level;
+import org.catrobat.catroid.uiespresso.util.rules.BaseActivityInstrumentationRule;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.Arrays;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+
+import static org.catrobat.catroid.uiespresso.content.brick.utils.BrickDataInteractionWrapper.onBrickAtPosition;
+import static org.catrobat.catroid.uiespresso.content.brick.utils.ColorSeekbarWrapper.MAX_COLOR_SEEKBAR_VALUE;
+import static org.catrobat.catroid.uiespresso.content.brick.utils.ColorSeekbarWrapper.MIN_SEEKBAR_VALUE;
+import static org.catrobat.catroid.uiespresso.content.brick.utils.ColorSeekbarWrapper.onColorSeekbar;
+
+@Category({Cat.AppUi.class, Level.Smoke.class})
+@RunWith(Parameterized.class)
+public final class ColorPickerParameterizedTest {
+	private static final Integer INIT_COLOR_VALUE = 128;
+	@Rule
+	public BaseActivityInstrumentationRule<SpriteActivity> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(SpriteActivity.class, SpriteActivity.EXTRA_FRAGMENT_POSITION, SpriteActivity.FRAGMENT_SCRIPTS);
+
+	@Before
+	public void setUp() {
+		createProject();
+		baseActivityTestRule.launchActivity();
+	}
+
+	@Parameterized.Parameters(name = "{3}")
+	public static Iterable<Object[]> data() {
+		return Arrays.asList(new Object[][] {
+				{R.id.color_rgb_seekbar_red,
+						R.id.brick_set_pen_color_action_red_edit_text,
+						(int) MAX_COLOR_SEEKBAR_VALUE,
+						"SeekbarRed"},
+				{R.id.color_rgb_seekbar_green,
+						R.id.brick_set_pen_color_action_green_edit_text,
+						(int) MIN_SEEKBAR_VALUE,
+						"SeekbarGreen"},
+				{R.id.color_rgb_seekbar_blue,
+						R.id.brick_set_pen_color_action_blue_edit_text,
+						INIT_COLOR_VALUE,
+						"SeekbarBlue"},
+		});
+	}
+
+	@Parameterized.Parameter
+	public @IdRes int colorRgbSeekbarId;
+
+	@Parameterized.Parameter(1)
+	public @IdRes int brickActionEditTextId;
+
+	@Parameterized.Parameter(2)
+	public int initValue;
+
+	@Parameterized.Parameter(3)
+	public String testName;
+
+	@Test
+	public void testColorSeekBar() {
+		int whenBrickPosition = 0;
+		onBrickAtPosition(whenBrickPosition).checkShowsText(R.string.brick_when_started);
+		int penColorBrickPosition = 1;
+
+		onBrickAtPosition(penColorBrickPosition).onFormulaTextField(brickActionEditTextId)
+				.checkShowsNumber(initValue)
+				.perform(click());
+
+		onView(withId(colorRgbSeekbarId))
```

Comment:
why swipe both directions? if the initial value is max, and the swipe action or the seekbar is broken, this test wount fail.
set the seekbar to 0 in the beginning of the test, then swipe one time (!) to right and verify its max.